libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Action/CSFSalesforceAction.m

```diff
@@ -323,20 +292,23 @@ + (BOOL)isNetworkError:(nullable NSError *)error {
     }
 }
 
-#pragma mark SFAuthenticationManagerDelegate
+- (void)didChangeUserDataNotification:(NSNotification*)notification {
+    CSFNetwork *network = (CSFNetwork*)notification.object;
+
+    if ([network isKindOfClass:[CSFNetwork class]]) {
+        SFUserAccount *userAccount = notification.userInfo[SFUserAccountManagerUserChangeUserKey];
+        SFUserAccountChange change = (SFUserAccountChange)[notification.userInfo[SFUserAccountManagerUserChangeKey] integerValue];
 
-- (void)userAccountManagerDidChangeCurrentUser:(NSNotification*)notification {
-    SFUserAccountManager *accountManager = (SFUserAccountManager*)notification.object;
-    if ([accountManager isKindOfClass:[SFUserAccountManager class]]) {
-        if (![accountManager.currentUserIdentity isEqual:self.enqueuedNetwork.account.accountIdentity]) {
-            self.enqueuedNetwork.networkSuspended = YES;
-        } else {
-            [self.enqueuedNetwork resetSession];
-            self.enqueuedNetwork.networkSuspended = NO;
+        [self willChangeValueForKey:@"isReady"];
+        if (change & SFUserAccountDataChangeCommunityId) {
+            self.enqueuedNetwork.defaultConnectCommunityId = userAccount.communityId;
         }
-        if (accountManager.currentCommunityId != self.enqueuedNetwork.defaultConnectCommunityId) {
-            self.enqueuedNetwork.defaultConnectCommunityId = accountManager.currentCommunityId;
+        SFUserAccountDataChange credsChanged = SFUserAccountDataChangeInstanceURL | SFUserAccountDataChangeAccessToken;
+        if ((change & credsChanged) == credsChanged) {
+            self.credentialsReady = YES;
```

Comment:
@khawkins the only other changes I see here are ID data & communityId? Should they affect the readiness of the operation?